hr_attendance_ip_check/models/hr_attendance.py

```diff
@@ -0,0 +1,35 @@
+import logging
+from odoo import models, api, _
+
+_logger = logging.getLogger(__name__)
+
+
+class HrAttendance(models.Model):
+    _inherit = 'hr.attendance'
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        """Override create to validate IP before creating attendance records."""
+        if not vals_list:
+            return {'warning': _('No valid attendance records to create')}
```

Comment:
No need to return warning for missing vals_list , it would be there. Anyways warning cannot be used here like this